blocks/algorithms/__init__.py

```diff
@@ -266,6 +268,8 @@ def __init__(self, cost=None, parameters=None, step_rule=None,
         # Set initial values for cost, parameters, gradients.
         self.cost = cost
         self.parameters = parameters
+        if gradients is not None and not isinstance(gradients, Mapping):
```

Comment:
Do you need to check if it is `Mapping`? Can gradients just always be converted to `OrderedDict`?